MoreLinq.Test/FallbackIfEmptyTest.cs

```diff
@@ -39,6 +39,28 @@ public void FallbackIfEmptyWithEmptySequence()
         }
 
         [Test]
+        public void FallbackIfEmptyPreservesSourceCollectionIfPossible()
+        {
+            var source = new int[] { 1 };
+            // ReSharper disable PossibleMultipleEnumeration
+            Assert.AreSame(source.FallbackIfEmpty(12), source);
```

Comment:
I did this on ccca164 , but I'm not sure it ended up being better, as it is a bunch of extra code noise.